x-pack/test/functional/services/cases/common.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { WebElementWrapper } from 'test/functional/services/lib/web_element_wrapper';
+import uuid from 'uuid';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function CasesAppCommonServiceProvider({ getService, getPageObject }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+  const find = getService('find');
+  const comboBox = getService('comboBox');
+  const header = getPageObject('header');
+  return {
+    /**
+     * Opens the create case page pressing the "create case" button.
+     *
+     * Doesn't do navigation. Only works if you are already inside a cases app page.
+     * Does not work with the cases flyout.
+     */
+    async openCreateCasePage() {
+      await testSubjects.click('createNewCaseBtn');
+      await testSubjects.existOrFail('create-case-submit', {
+        timeout: 5000,
+      });
+    },
+
+    /**
+     * it creates a new case from the create case page
+     * and leaves the navigation in the case view page
+     *
+     * Doesn't do navigation. Only works if you are already inside a cases app page.
+     * Does not work with the cases flyout.
+     */
+    async createCaseFromCreateCasePage(caseTitle: string = 'test-' + uuid.v4()) {
```

Comment:
I don't like the idea of having randomized titles as it is very difficult to test the text of the title.

Reply:
This is necessary because many tests rely on using unique test titles to test different things.  You can overwrite the case title for test cases where you need an specific one.

Reply:
Is this only being used in `create_case_form`, correct? If a consumer of the API uses `createCaseFromCreateCasePage()` and then want to check if the case is created by looking at the title she cannot do that because the title is not returned. So we are always forcing them to put a title. I am ok with the function as it is. I just wanted to raise this concern.

Reply:
@cnasikas in such case they'd be using the API incorrectly. If they want to have the case title, they should pass it instead of expect this function to return it. 

In any case. I initially thought this method was going to be more widely used but it ended up in a single test. Maybe in the future we use it more.